src/foam/nanos/notification/email/EmailTemplateUtil.js

```diff
@@ -0,0 +1,396 @@
+            return val;
+          }
+        };
+        grammar.addAction("ANY_CHAR", anyCharAction);
+
+
+        // simple value syntax: "qwerty {{ simple_value }} qwerty"
+        grammar.addSymbol("SIMPLE_VAL", new Seq1(2, Literal.create("{{"), Whitespace.instance(),
+          new Repeat0(new AnyKeyParser()), Whitespace.instance() ,Literal.create("}}") ));
+        Action simpleValAction = new Action() {
+          @Override
+          public Object execute(Object val, ParserContext x) {
+            String value = (String) getValues().get(val);
+            if ( value == null ) {
+              value = "undefined";
+              foam.nanos.logger.Logger logger = (foam.nanos.logger.Logger) x_.get("logger");
+              logger.error("No value provided for variable " + val);
+            }
+            getSb().append(value);
+            return value;
+          }
+        };
+        grammar.addAction("SIMPLE_VAL", simpleValAction);
+
+
+        /* IF_ELSE syntax: "qwerty {% if var_name_provided_in_map %} qwer {{ possible_simple_value }} erty
+        {% else %} qwerty {% endif %}" */
+        Parser ifElseParser = new SeqI(new int[] { 4, 7, 13 },
+          Literal.create("{%"),
+          Whitespace.instance(),
+          Literal.create("if"),
+          Whitespace.instance(),
+          new Repeat(new Until(Literal.create("%}"))),
+          Whitespace.instance(),
+          Literal.create("%}"),
+          (new Repeat(new Until(Literal.create("{%")))),
+          Literal.create("{%"),
+          Whitespace.instance(),
+          Literal.create("else"),
+          Whitespace.instance(),
+          Literal.create("%}"),
+          (new Repeat(new Until(Literal.create("{%")))),
+          Literal.create("{%"),
+          Whitespace.instance(),
+          Literal.create("endif"),
+          Whitespace.instance(),
+          Literal.create("%}")
+        );
+
+        grammar.addSymbol("IF_ELSE", ifElseParser);
+        Action ifElseAction  = new Action() {
+          @Override
+          public Object execute(Object val, foam.lib.parse.ParserContext x) {
+            Object[] valArr = (Object[]) val;
+            StringBuilder ifCond = new StringBuilder();
+            Object[] val0 = (Object[]) valArr[0];
+            for ( int i = 0 ; i < val0.length ; i++ ) {
+              if ( ! Character.isWhitespace((char)val0[i]) ) ifCond.append(val0[i]);
+            }
+
+            StringBuilder finalVal = new StringBuilder();
+            if ( getValues().get(ifCond.toString() ) != null ) {
+              Object[] val1 = (Object[]) valArr[1];
+              for ( int i = 0 ; i < val1.length ; i++ ) {
+                finalVal.append(val1[i]);
+              }
+            } else {
+              Object[] val2 = (Object[]) valArr[2];
+              for ( int i = 0 ; i < val2.length ; i++ ) {
+                finalVal.append(val2[i]);
+              }
+            }
+            StringPStream finalValPs = new StringPStream();
```

Comment:
What are you doing here?

Reply:
I'm re-parsing the inner part in case there is other symbols inside if-else statements